packages/jsonapi/middleware.js

```diff
@@ -183,27 +183,17 @@ class Handler {
       sort: this.query.sort,
       page: this.query.page,
       queryString: this.query.q
-    });
+      // TODO: I feel like we can do better here....
+    }, flatten(this.includes || []));
```

Comment:
@ef4 I'm not so happy about this either as I'm collapsing path information by flattening this....